R/expect-condition.R

```diff
@@ -106,6 +110,25 @@ expect_condition <- function(object,
   invisible(act$val %||% act$cap)
 }
 
+capture_first_condition <- function(expr) {
+  first <- NULL
+
+  tryCatch(
+    withCallingHandlers(expr, condition = function(cnd) {
+      if (is.null(first)) {
+        if (can_entrace(cnd)) {
+          cnd <- cnd_entrace(cnd)
+        }
+        first <<- cnd
+      }
+      cnd_muffle(cnd)
+    }),
+    error = function(cnd) {}
```

Comment:
Why capture errors? I would expect `expect_condition()` to let errors abort. Are we missing a test about this? e.g. 

```r
expect_error(
  expect_condition({ message("foo"); stop("bar") }),
  "bar"
)
```

Reply:
Oh I just assumed that since errors were conditions they should be caught here. Probably we should just be consistent with however it worked before.

Reply:
oh I think they should be captured if they are the first condition, but we shouldn't recover from ulterior errors.

Reply:
Ok, please take another look — I wasn't sure if there was a better way to "muffle" the first error, but this seems to work (and I added more tests along the lines you suggested).